test.mjs

```diff
@@ -0,0 +1,4022 @@
+import {beforeEach, describe, it} from 'mocha';
+import parser from './index.mjs';
+
+import * as chai from 'chai';
+chai.should()
+
+// const { expect } = require('chai')
+// const fs = require('fs')
+// const parser = require('../build/index.cjs')
+// const path = require('path')
+
+describe('yargs-parser', function () {
+  it('should parse a "short boolean"', function () {
+    const parse = parser(['-b'])
+    parse.should.not.have.property('--')
+    parse.should.have.property('b').to.be.ok.and.be.a('boolean')
+    parse.should.have.property('_').with.length(0)
+  })
+
+  it('should parse a "long boolean"', function () {
+    const parse = parser('--bool')
+    parse.should.not.have.property('--')
+    parse.should.have.property('bool', true)
+    parse.should.have.property('_').with.length(0)
+  })
+
+  
+  it('should place bare options in the _ array', function () {
+    const parse = parser('foo bar baz')
+    parse.should.have.property('_').and.deep.equal(['foo', 'bar', 'baz'])
+  })
+
+  it('should set the value of the final option in a group to the next supplied value', function () {
+    const parse = parser(['-cats', 'meow'], {
+      withValue: ['s']
```

Comment:
Most significant API change so far, is that we must set `withValue` for any arguments that expect a value.